### PR TITLE
[ENH] Add `is_weekend` option to `DateTimeFeatures` trafo

### PR DESCRIPTION
**Pull request by KishManani**
#### What does this implement/fix? Explain your changes.
This PR adds `is_weekend` as an option to the `manual_selection` argument in the `DateTimeFeatures` transformer. I have also added more examples to the docstring.

Example usage:
```Python
import pandas as pd
from sktime.transformations.series.date import DateTimeFeatures

df = pd.DataFrame(
        data={"y": [1, 1, 1, 1, 1, 1, 1]},
        index=pd.date_range(start="2000-01-01", freq="D", periods=7),
    )

result = DateTimeFeatures(manual_selection=["is_weekend"]).fit_transform(df)

```

#### What should a reviewer concentrate their feedback on?
The changes made to `_RAW_DUMMIES` on line 32. I'm forced to create a child/parent relationship so I made the child `day` and parent `week`. I'm unsure whether this could have unintended consequences. None were picked up in testing.

The setting of the `fourier` column in the `DUMMIES` dataframe on line 324. I am unsure whether this is the correct value to set for the `fourier` column and what impact this might have?

##### For all contributions
- [x] I've added unit tests and made sure they pass locally.
- [x] The PR title starts with either [ENH], [MNT], [DOC], or [BUG] indicating whether the PR topic is related to enhancement, maintenance, documentation, or bug.

### CODE REVIEW

**Review by fkiraly — APPROVED** [review]
looks good to me!

Before we merge this, @danbartl needs to approve whether this change is ok, since by default he owns the estimator.

Odd that he @danbartl was not auto-requested, are you not in the codeowners?

**Comment by KishManani** [comment]
Apologies @fkiraly! I corrected a typo in a doc string and it needs to be approved again :-/

**Review by fkiraly — APPROVED** [review]
sure

**Comment by danbartl** [comment]
> The setting of the fourier column in the DUMMIES dataframe on line 324. I am unsure whether this is the correct value to set for > the fourier column and what impact this might have?

The Fourier thing is fine, I will remove it in the next release. The logic to add it as day in week makes sense.

**Pull request merged**
